src/Maker/MakeTest.php

```diff
@@ -106,14 +108,40 @@ public function interact(InputInterface $input, ConsoleStyle $io, Command $comma
             if (!isset(self::DESCRIPTIONS[$type])) {
                 throw new RuntimeCommandException(sprintf('The test type must be one of "%s", "%s" given.', implode('", "', array_keys(self::DESCRIPTIONS)), $type));
             }
+        } else {
+            $input->setArgument(
+                'type',
+                $io->choice('Which test type would you like?', self::DESCRIPTIONS)
+            );
+        }
 
-            return;
+        if ('ApiTestCase' === $input->getArgument('type') && !class_exists(ApiTestCase::class)) {
+            $io->warning([
+                'API Platform is required for this test type. Install it with',
+                'composer require api',
+            ]);
         }
 
-        $input->setArgument(
-            'type',
-            $io->choice('Which test type would you like?', self::DESCRIPTIONS)
-        );
+        if ('PantherTestCase' === $input->getArgument('type') && !trait_exists(PantherTestCaseTrait::class)) {
+            $io->warning([
+                'symfony/panther is required for this test type. Install it with',
+                'composer require symfony/panther --dev',
+            ]);
+        }
+
+        if (null === $input->getArgument('name')) {
+            $io->writeln([
+                '',
+                'Choose a class name for your test, like:',
+                ' * <fg=yellow>UtilTest</> (to create tests/UtilTest.php)',
+                ' * <fg=yellow>Service\\UtilTest</> (to create tests/Service/UtilTest.php)',
+                ' * <fg=yellow>\\App\Tests\\Service\\UtilTest</> (to create tests/Service/UtilTest.php)',
```

Comment:
Could we (should we) suggest the class based on the test? e.g. if the type is `UnitTest` - we suggest `'App\\Tests\\UnitTests\\{NAME}` or something along those lines?

Reply:
I thought about that too. Let's save it for a future PR... I don't think there is clear agreement on the directory conventions of these things. I personally don't sub-divide things into `UnitTest` or `IntegrationTest` directories, but I know that some people do this.